src/eei.cpp

```diff
@@ -206,6 +218,7 @@ namespace HeraVM {
 
       HERA_DEBUG << "callDataCopy " << hex << resultOffset << " " << dataOffset << " " << length << dec << "\n";
 
+      takeGas(GasSchedule::stepGas2 + GasSchedule::copy * (length / 32));
```

Comment:
The rules applied in `codeCopy` and `externalCodeCopy` are not really followed here?